pkg/subctl/cmd/diagnose/globalnet.go

```diff
@@ -0,0 +1,162 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package diagnose
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd"
+	"github.com/submariner-io/submariner/pkg/globalnet/constants"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcsClientset "sigs.k8s.io/mcs-api/pkg/client/clientset/versioned"
+)
+
+func init() {
+	diagnoseCmd.AddCommand(&cobra.Command{
+		Use:   "globalnet",
+		Short: "Check globalnet configuration",
+		Long:  "This command checks globalnet configuration",
+		Run: func(command *cobra.Command, args []string) {
+			cmd.ExecuteMultiCluster(restConfigProducer, checkGlobalNet)
+		},
+	})
+}
+
+func checkGlobalNet(cluster *cmd.Cluster) bool {
+	status := cli.NewStatus()
+
+	if cluster.Submariner == nil {
+		status.Start(cmd.SubmMissingMessage)
+		status.EndWith(cli.Warning)
+
+		return true
+	}
+
+	status.Start("Checking Globalnet configuration")
+
+	if cluster.Submariner.Spec.GlobalCIDR != "" {
+		checkClusterGlobalEgressIps(cluster, status)
+		checkGlobalEgressIps(cluster, status)
+		checkGlobalIngressIps(cluster, status)
+
+		if status.HasFailureMessages() {
+			status.EndWith(cli.Failure)
+			return false
+		}
+
+		status.EndWithSuccess("Globalnet is enabled and properly configured")
+	} else {
+		status.EndWithSuccess("Globalnet is disabled")
+	}
+
+	return true
+}
+
+func checkClusterGlobalEgressIps(cluster *cmd.Cluster, status *cli.Status) {
+	clusterGlobalEgress, err := cluster.SubmClient.SubmarinerV1().ClusterGlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining the clusterGlobalEgressIPs cr: %v", err))
+		return
+	}
+
+	if len(clusterGlobalEgress.Items) != 1 {
+		status.QueueFailureMessage(fmt.Sprintf("Number (%d) of clusterGlobalEgress resources != 1", len(clusterGlobalEgress.Items)))
+		return
```

Comment:
I think this can be a warning since we only process the default instance. Also don't return here.

```suggestion
		status.Warning("Found %d ClusterGlobalEgressIP resources but only the default instance is supported", len(clusterGlobalEgress.Items))
```

Reply:
We can also print the name of the default instance (i.e., "cluster-egress.submariner.io") to make it easy for the user.